packages/default-theme/components/checkout/steps/ShippingStep.vue

```diff
@@ -86,77 +74,23 @@ export default {
+    const { shippingMethod, setShippingMethod } = useSessionContext()
+    const { refreshCart } = useCart()
 
-    return {
-      isGuestOrder,
-      shippingMethods,
-      shippingMethod,
-    }
-  },
-  data() {
-    return {
-      shippingMethod: '',
-      shippingMethods: [
-        {
-          isOpen: false,
-          price: 'Free',
-          delivery: 'Delivery from 3 to 7 business days',
-          name: 'Pickup in the store',
-          id: 'store',
-          translated: {
-            description:
-              'Novelty! From now on you have the option of picking up an order in the selected InPack parceled. Just remember that in the case of orders paid on delivery, only the card payment will be accepted.',
-          },
-        },
-        {
-          isOpen: false,
-          price: '$9.90',
-          delivery: 'Delivery from 4 to 6 business days',
-          name: 'Delivery to home',
-          id: 'home',
-          translated: {
-            description:
-              'Novelty! From now on you have the option of picking up an order in the selected InPack parceled. Just remember that in the case of orders paid on delivery, only the card payment will be accepted.',
-          },
-        },
-        {
-          isOpen: false,
-          price: '$9.90',
-          delivery: 'Delivery from 4 to 6 business days',
-          name: 'Paczkomaty InPost',
-          id: 'inpost',
-          translated: {
-            description:
-              'Novelty! From now on you have the option of picking up an order in the selected InPack parceled. Just remember that in the case of orders paid on delivery, only the card payment will be accepted.',
-          },
-        },
-        {
-          isOpen: false,
-          price: '$11.00',
-          delivery: 'Delivery within 48 hours',
-          name: '48 hours coffee',
-          id: 'coffee',
-          translated: {
-            description:
-              'Novelty! From now on you have the option of picking up an order in the selected InPack parceled. Just remember that in the case of orders paid on delivery, only the card payment will be accepted.',
-          },
-        },
-        {
-          isOpen: false,
-          price: '$14.00',
-          delivery: 'Delivery within 24 hours',
-          name: 'Urgent 24h',
-          id: 'urgent',
-          translated: {
-            description:
-              'Novelty! From now on you have the option of picking up an order in the selected InPack parceled. Just remember that in the case of orders paid on delivery, only the card payment will be accepted.',
-          },
-        },
-      ],
-    }
+    const activeShippingMethod = computed({
+      get: () => shippingMethod.value && shippingMethod.value.id,
+      set: async (id) => {
+        await setShippingMethod({ id })
+        await refreshCart()
+      },
+    })
+
+    onMounted(async () => {
+      await getShippingMethods()
```

Comment:
since we are not doing anything with the response here, we don't need to await inside.

Reply:
a good question :) it's just to show it's a promise and someone can do some action after it's resolved; it doesn't harm anyone and not slow anything down :)